### ISSUE TITLE
alerting.telegram.overrides seems to be broken

### ISSUE THREAD

**Issue by baalajimaestro** [TwiN/gatus#919]
### Describe the bug

When i add alerting.telegram.overrides, the provider configuration becomes invalid

```
    overrides:
      - group: "something" // redacted
        id: "1100" // redacted
```

Remove the entire override, and it becomes functional again

### What do you see?

2024/11/30 16:33:36 [config.validateAlertingConfig] Ignoring provider=telegram because configuration is invalid

### What do you expect to see?

working provider

### List the steps that must be taken to reproduce this issue

1. add overrides with either token or id or both to telegram provider, it breaks.

### Version

latest tag on docker.

### Additional information

It would be also helpful if the logs said why it is invalid, instead of a blind mention of this config is invalid, thats neither helpful, and wastes a ton of time with trial and error.

**Comment by lonesomewalker** [comment]
Well, as a "should work" project, i did not expect to see something like this stale for 2 weeks now.
Installed it, SMTP broken, Telegram broken.

Whoa, and i thought i found something good.

**Comment by TwiN** [comment]
@baalajimaestro FYI, I'm currently working on rewriting a big part of how overrides work in Gatus, which does fix this issue as well as address the comments you mentioned in your additional notes wrt errors lacking visibility, see #929

---

@lonesomewalker Thanks for your input, but I have a job and a family to take care of, I don't work on Gatus full time because it doesn't pay the bills. Yet, I try my best to provide useful tools for developers during my personal time, and sarcastic comments like yours certainly don't motivate open source maintainers.

**Comment by lonesomewalker** [comment]
> @baalajimaestro FYI, I'm currently working on rewriting a big part of how overrides work in Gatus, which does fix this issue as well as address the comments you mentioned in your additional notes wrt errors lacking visibility, see #929

Then a short notify to the person wo reports this would be VERY nice, so i would also not have spent MY free time to also try to find the error for myself :-)

> @lonesomewalker Thanks for your input, but I have a job and a family to take care of, I don't work on Gatus full time because it doesn't pay the bills. Yet, I try my best to provide useful tools for developers during my personal time, and sarcastic comments like yours certainly don't motivate open source maintainers.

We ALL do this on certain projects. But it is a matter of how we deal with input from others.
I agree that AI bullshit is overwhelming, also feedback from users who only say "give for free", but if somebody already sent a pull request to an issue which is somehow usefull... at least a VERY short feedback is something which is doable.

**Comment by TwiN** [comment]
To get back on topic, I've merged #929 to master yesterday. When you have some time, could you let me know if it fixes the issue? I haven't had time to test every alerting provider manually myself, but I was planning on letting this soak on master/latest for some time before releasing.

**Comment by lonesomewalker** [comment]
> Using opensource projects that don't come with SLA or corporate backing means that people like @TwiN and me are doing this on our free will, as a hobby. It's very obvious we aren't getting paid for this. You can bitch about this all you want. You could have used your free time to actually look at why that PR was open for 2 weeks. It was literally my fault, since my editor replaced all the tabs with spaces, which broke the linter.

Github is full of shit. A lot of projects which are abandoned now.
And i don't want to waste my time with such.

So, tell me, where is the flame?

> I get that you're mad about debugging this error for hours - I did that too. But flaming the project owner for something he had no control over is just pathetic. You could have looked at the comments right on that PR to see why it was stalled. Instead, you chose to come here and flame the project owner for not communicating whatever "you deem necessary". You're contributing nothing here, just reducing morale by flaming people when all you had to do was READ. Not that hard, is it?

No, i am not mad about debugging, just the lack of information. Something which was working in the past.
A lot off issues are because of bad or no communication.

Also yours. :-)
What i give and where is a) none of your business and b) @TwiN can assure you, i did.
So better keep your mouth shut.

**Comment by baalajimaestro** [comment]
> To get back on topic, I've merged #929 to master yesterday. When you have some time, could you let me know if it fixes the issue? I haven't had time to test every alerting provider manually myself, but I was planning on letting this soak on master/latest for some time before releasing.

This works pretty fine for Telegram!
